### ISSUE TITLE
Neo4J docker not starting with custom user and mounted directories

### ISSUE THREAD

**Issue by rfriesen-vt** [neo4j/docker-neo4j#351]
Operating System: MacOS 12.4
Neo4J docker image: 4.4.4+
docker version: 20.10.16, build aa7e414

Expected behaviour:
- Neo4J starts inside the docker container successful

Actual behaviour:
- Docker container crashes with error: 
`Consider unsetting SECURE_FILE_PERMISSIONS environment variable, to enable docker to write to /logs.

Folder /logs is not accessible for user: 503 or group 20 or groups 20, this is commonly a file permissions issue on the mounted folder.

Hints to solve the issue:
1) Make sure the folder exists before mounting it. Docker will create the folder using root permissions before starting the Neo4j container. The root permissions disallow Neo4j from writing to the mounted folder.
2) Pass the folder owner's user ID and group ID to docker run, so that docker runs as that user.
If the folder is owned by the current user, this can be done by adding this flag to your docker run command:
  --user=$(id -u):$(id -g)`

Steps to reproduce:
Start Neo4j-docker with version 4.4.4 or above using:
`docker run \
    --user $(id -u ${USER}):$(id -g ${USER}) \
    --publish=7474:7474 --publish=7687:7687 \
    --volume="$BASEPATH/neo4j/data":/data \
    --volume="$BASEPATH/neo4j/logs":/logs \
    --volume="$BASEPATH/../import/csv":/import \
    --volume="$BASEPATH/conf":/conf \
    neo4j:4.4.4`

where $BASEPATH refers to the current directory incl. the given folders for logs, data and import with write privileges set for the current user as owner permissions

Root cause:
There was a change in the "docker-entrypoint.sh" script as of version 4.4.4 now using the "is_writable" function inside "check_mounted_folder_writable_with_chown". According to my investigation the owner of the folders inside the container is set to "root" even though the --user flag is passed to the container start. That leads to a mismatch when comparing the current user and the folder owner in the "is_writable" function.

**Comment by jennyowen** [comment]
This is really weird.
If I start a `neo4j:4.4.4` container on Mac and `stat /logs` it shows the owner as `root` and permissions as 755, so of course the `is_writable` check returns false... but then if I `ls /logs` then (after a few tries) `stat /logs` returns the folder owner as 501??

My guess is that the docker desktop VM is introducing some kind of delay or has lazy file permissions that is causing `stat` to lie about the correct folder owner until you actually look inside the folder.
That makes me wonder how we were **ever** able to check file permissions correctly. I had a look to see what changed between 4.4.3 and 4.4.4, and it used to be that we used `test -w $file` to check for writability. If I try `test -w /logs` inside the container then it returns true. 

I will change the writability check in docker-entrypoint.sh to use `test -w`, but I can't fix the underlying problem that Docker on Mac is doing some weird stuff to file/folder permissions.

**Comment by szarnyasg** [comment]
Thanks, it now works for me, I tried Neo4j versions 4.4.4 and 4.4.11 (MacOS 12.6, M1 CPU).

**Comment by szarnyasg** [comment]
Sorry, I take it back: 4.4.11 and 4.4.12 with a mounted `/data` directory and it fails with:
```console
Consider unsetting SECURE_FILE_PERMISSIONS environment variable, to enable docker to write to /data.

Folder /data is not accessible for user: 501 or group 20 or groups 20, this is commonly a file permissions issue on the mounted folder.
```

**Comment by jennyowen** [comment]
@szarnyasg ah sorry that's my mistake for naming the PR badly. 
I've fixed the **source code** for our docker image but I haven't pushed the updated docker images to docker hub. So unless you build the image yourself, it won't be fixed for you yet.

Generally I don't like to apply fixes to images that we've already released because people don't like things being changed underneath them and I don't want to risk introducing bugs via a docker cache update.

Just looking at our release schedule, 4.4.13 should be out some time next week, and that will have the relevant fix. There should also be a 5.1.0 release next week and that will have the fix as well.

**Comment by szarnyasg** [comment]
@jennyowen Excellent, having the fix available next week works perfectly for me. I'll wait and test 4.4.13 then and let you know the result.

**Comment by jennyowen** [comment]
@szarnyasg it's looking like 4.4.13 won't make it out this week unfortunately. We did release 5.1.0 though, so you might have luck there.

**Comment by szarnyasg** [comment]
4.4.13 is out and fixed this issue for me on MacOS 13.0.

**Comment by jennyowen** [comment]
@szarnyasg thanks for the update, I'm glad it works!
